### PR TITLE
Rewrote AutoSmelter

### PR DESCRIPTION
**Pull request by Thunderkill**
**AutoSmelter had a few issues:**
- Using unwanted items as fuel (for example enchanted bows)
- Smelting unwanted items (for example cobblestone)
- Not taking out items before changing smeltable item

These are the issues i planned to fix, but happened to rewrite almost the whole thing.

**New features:**
You select what items to smelt and what items to use as fuel
Will take out smelted items immediately
Ability whether to disable the module when you run out of items or not

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSmelter.java [hunk 1]
```diff
@@ -6,137 +6,125 @@
...
+    private final Setting<Boolean> disableWhenOutOfItems = sgGeneral.add(new BoolSetting.Builder()
+        .name("disable-when-out-of-items")
+        .description("Disable the module when you run out of items")
+        .defaultValue(true)
+        .build()
+    );
+
 
     public AutoSmelter() {
         super(Categories.World, "auto-smelter", "Automatically smelts all items in your inventory that can be smelted.");
```
Comment (timoreo22) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSmelter.java:55]: This description should probably be changed

**Review by timoreo22 — CHANGES_REQUESTED** [review]
Looks good ! the module's description should probably be changed tho

File: src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSmelter.java (hunk 1 above)
Comment (Thunderkill) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSmelter.java:55]: I shall change it immediately

**Comment by Thunderkill** [comment]
There seems to be some crashing on startup, i'll need to investigate that for a bit.

**Comment by Thunderkill** [comment]
Also, the game will crash if you try to open the settings while in main menu, any ideas how i could fix that?

It happens because it is checking for all items that are smeltable, but that can't be done in the menu i assume.

**Comment by MineGame159** [comment]
Just don't display anything while in main menu

**Comment by Thunderkill** [comment]
> Just don't display anything while in main menu

Or then display all items?

**Comment by MineGame159** [comment]
I think it would be better to not display anything

**Comment by Thunderkill** [comment]
Closing PR in favor of a new one

**Pull request closed without merging**